### PR TITLE
fix: update cudaMemAdvise host location

### PR DESCRIPTION
**Pull request by camirian**
## Summary
- update the managed-allocation fallback to pass a cudaMemLocation host target to cudaMemAdvise
- fixes compatibility with CUDA headers where cudaMemAdvise expects struct cudaMemLocation instead of an int device id

## Verification
- git diff --check
- C++ syntax-only check against local CUDA 13 headers: old cudaCpuDeviceId argument fails, new cudaMemLocation{cudaMemLocationTypeHost, 0} form passes

## Not run
- full nvcc build/test; this host does not have a usable x86_64 nvcc available

### CODE REVIEW

**Comment by camirian** [comment]
EdgeXpert verification update:

- Host: MSI EdgeXpert / NVIDIA GB10 / aarch64.
- Driver reports CUDA 13.0; installed toolkit used for build: CUDA 13.2 (`/usr/local/cuda-13.2/bin/nvcc`, V13.2.78).
- `make test_gpt2cu PRECISION=FP32 NVCC=/usr/local/cuda-13.2/bin/nvcc` compiled successfully, but the produced binary failed at runtime with `the provided PTX was compiled with an unsupported toolchain` because CUDA 13.2 PTX was too new for the installed driver.
- Rebuilt native SASS for GB10 compute capability 12.1 with:

```bash
make -B test_gpt2cu PRECISION=FP32 NVCC=/usr/local/cuda-13.2/bin/nvcc NVCC_FLAGS="--threads=0 -t=0 --use_fast_math -std=c++17 -O3 --generate-code=arch=compute_121,code=sm_121"
```

- That build succeeded and `./test_gpt2cu` ran on `NVIDIA GB10`.
- Runtime did not fully pass: logits/loss checks showed small FP32 mismatches on GB10 (`logit max diff: 0.001022`, later loss mismatches at steps 8-10), then the test stopped on missing `dev/data/tinyshakespeare/tiny_shakespeare_val.bin`.

So the PR is verified to compile against CUDA 13.2 headers/toolchain and run into model execution on GB10 when compiled for native `sm_121`, but I am keeping it draft until a clean target-platform test pass is available.

**Comment by camirian** [comment]
Closing this draft. This was an exploratory EdgeXpert/CUDA 13 compatibility probe and is no longer part of my active work. Thanks.
